File: 9-palindrome-number/palindrome-number.c
bool isPalindrome(int x) {
    long original = x;
    long result = 0;
     if( x < 0)
        return false;

    while( x>0){
        long last = x % 10;
        result = result *10 +last ;
        x /= 10;
    }
    return original == result;
    
}

File: 9-palindrome-number/palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        if (x < 0) return false;
        
        int original = x;
        int result = 0;

        while (x > 0) {
            int last = x % 10;  
            result = result * 10 + last; 
            x /= 10;
        }
        
        return original == result;
    }
}
